### ISSUE TITLE
GetPrettyUpdateURL Page Not Found

### ISSUE THREAD

**Issue by OSPete** [kfrancis/chargify-dot-net#54]
The GetPrettySubscriptionUpdateURL is 404'ing because the "/" parts of the URL itself are getting encoded. I haven't been able to narrow down what has changed on chargify's end to cause this - it does seem like a bug in Chargify.NET though. I've outlined fixes below.

If you update the [Chargify.NET ChargifyConnect](https://github.com/kfrancis/chargify-dot-net/blob/3ae0454ebeaa4e074c2ae2378f628ccba60926c7/Source/ChargifyDotNet/ChargifyConnect.cs#L1159) class to the following it should resolve the issue. 
```
        public string GetPrettySubscriptionUpdateURL(string firstName, string lastName, int subscriptionId)
        {
            if (string.IsNullOrEmpty(SharedKey)) throw new ArgumentException("SharedKey is required to generate the hosted page url");

            string message = UpdateShortName + "--" + subscriptionId + "--" + SharedKey;
            string token = message.GetChargifyHostedToken();
            string prettyId = string.Format("{0}-{1}-{2}", subscriptionId, PCLWebUtility.WebUtility.UrlEncode(firstName.Trim().ToLower()), PCLWebUtility.WebUtility.UrlEncode(lastName.Trim().ToLower()));
            string methodString = string.Format("{0}/{1}/{2}", UpdateShortName, prettyId, token);
            // just in case?
            // methodString = PCLWebUtility.WebUtility.UrlEncode(methodString);
            string updateUrl = string.Format("{0}{1}{2}", URL, (URL.EndsWith("/") ? "" : "/"), methodString);
            return updateUrl;
        }
```

All i've done is comment out the global Urlencode call and wrap first and last name in encodes instead (as I believe those are the 2 potential problem areas - each of the sections could be wrapped in encodes instead). I also believe the URL encode in [this method](https://github.com/kfrancis/chargify-dot-net/blob/3ae0454ebeaa4e074c2ae2378f628ccba60926c7/Source/ChargifyDotNet/ChargifyConnect.cs#L1178) could be removed or modified as deemed appropriate.

Edit: I've looked into making a pull request but There's a billion errors - i think related to nuget packages.. or something... that I can't seem to fix at this time. 

Edit 2: I updated the code snippets above - I was looking at the Chargify.NET folder in source - and that's apparently.. no longer.. needed/updated/used? I'm honestly not sure - but these changes are based on the code in the chargifydotnet structure instead. 

Edit 3: I defeated VS/Git/TheWorld - PR at #55

**Comment by kfrancis** [comment]
Nice! I'll look at this shortly.

**Comment by OSPete** [comment]
Just confirmed this morning the GetSubscriptionUpdateURL does have the same issue - we're using it in another area, so both of those changes would be needed. (Note: Again, you could just URLEncode each of the properties in the URL so the "/' isn't caught up in it - I just chose not to because first name/last name seems like the only potential areas for error)

**Comment by kfrancis** [comment]
Thanks @OSPete for the PR, merged!

**Comment by kfrancis** [comment]
Released in [version 1.2.3](https://www.nuget.org/packages/chargify/1.2.3).
